proxy_agent/src/shared_state.rs

```diff
@@ -72,6 +77,10 @@ impl SharedState {
 impl Default for SharedState {
     fn default() -> Self {
         SharedState {
+            runtime: Some(Arc::new(Mutex::new(
+                tokio::runtime::Runtime::new().unwrap(),
+            ))),
```

Comment:
I really wouldn't recommend this. Use the [tokio::main](https://docs.rs/tokio/latest/tokio/attr.main.html) macro.